docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,220 @@
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Create the model assertion inside the project directory.
+
+Set the following:
+
+
+- ``authority-id``, ``brand-id`` to your developer ID
+- ``timestamp`` to a RFC3339 formatted string, within you signing key's validity
+- ``store`` to your dedicated Snap Store ID
+
+Change the value of ``model`` to something representative of your model.
+
+The ``snaps`` array is a list of snaps that get included in the image.
+The gadget snap has no listed ``channel`` and ``id``, because it isn't in a Store.
+We have built it locally in the earlier steps and will later on pass it directly to the image builder.
+In practice, the gadget snap should be uploaded to a Store and then listed in the model assertion along with its channel and id.
+Uploading to the store makes it possible to use a signed snap that receives updates.
+
+The ``realtime-kernel`` snap contains the realtime Linux kernel.
+This snap should be included explicitly in your dedicated Snap Store.
+
+Add any other snaps, including for example your realtime application.
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+
+Here are the needed steps:
+
+1) Create and register a key
+
+
+.. code-block:: shell
+
+    snapcraft create-key realtime-ubuntu
+    snapcraft register-key realtime-ubuntu
+
+
+You can use ``snapcraft list-keys`` to check your existing keys.
+
+2) Sign the model assertion
+
+.. code-block:: shell
+
+    snap sign -k realtime-ubuntu model.json > model.signed.yaml
+
+The ``snap sign`` command takes JSON as input and produces YAML as output!
+
+.. tip::
+
+    You need to repeat the signing every time you change the input model, because the signature is calculated based on the model.
+
+Before we continue, let's have an overview of the files inside our project directory:
+
+.. code-block:: console
+
+    $ tree -L 1
+    .
+    ├── model.json
+    ├── model.signed.yaml
+    └── pc-gadget
+
+    2 directories, 2 files
+
+The project directory should contain the model assertion, the signed model assertion, and pc-gadget directory.
+
+Build the Ubuntu Core image
+---------------------------
+
+First, get familiar with the tooling by referring to the guide on `building Ubuntu Core images`_.
+
+We use ``ubuntu-image`` and need to set the paths to the following as input:
+
+- Exported store credentials
+- Signed model assertion YAML file
+- Locally built gadget snap
+
+Export the store credentials to a file:
+
+.. code-block:: shell
+
+    snapcraft export-login credentials.txt
+
+Then build the image:
+
+.. code-block:: console
+
+    $ UBUNTU_STORE_AUTH_DATA_FILENAME=credentials.txt \
```

Comment:
Why is this necessary? is it specific to real time?
The UC doc about creating your custom image doesn't seem to mention that.

Reply:
This is because the realtime-kernel snap is a commercial snap, available via dedicated snap stores only.